# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/bicon/03-hotfix-fribidi.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

diff --git a/bicon/pty_spawn.c b/bicon/pty_spawn.c
index 21ab737..220e632 100644
--- a/bicon/pty_spawn.c
+++ b/bicon/pty_spawn.c
@@ -93,7 +93,7 @@ _xwrite (
   }
 }
 
-static int
+static void
 _copy (
   int master_fd,
   reader master_read,
diff --git a/bjoining/bjoining_log2cuni.c b/bjoining/bjoining_log2cuni.c
index 7c2f54c..ec1e893 100644
--- a/bjoining/bjoining_log2cuni.c
+++ b/bjoining/bjoining_log2cuni.c
@@ -21,7 +21,7 @@ bjoining_log2cuni (
   if (0 == (options & (B_LOGICAL_OUTPUT | B_LOGICAL_OUTPUT_LOG2CUNI)))
     {
       f = f
-	&& fribidi_log2vis ((FriBidiChar *) str, len, &ptype,
+	&& fribidi_log2vis ((FriBidiChar *) str, len, (FriBidiParType * )&ptype,
 			    (FriBidiChar *) vis, NULL, NULL, NULL);
     }
   else
